src/main/scala/com/databricks/spark/xml/util/InferSchema.scala

```diff
@@ -48,6 +48,7 @@ private[xml] object InferSchema {
       FloatType,
       DoubleType,
       TimestampType,
+      DateType,
```

Comment:
Hm .. this actually needs a more fundamental fix. We should not state the precedence in this order (date is wider then timestamp). I am pretty sure this could lead to ending up with date during resolving the type conflicts between date and timestamp.

For example, 

```
"1999/01/11" -> date
"2017/05/15T14:58:19Z" -> timestamp
```

then it becomes date time which I guess will truncate the time part in the timestamp.